fix: `format_as("datasets")` when no responses

alvarobartt opened the pull request:
# Description

As just reported by @dvsrepo, the following flow: `FeedbackDataset.from_argilla() -> .format_as("datasets")` was failing under some scenarios where either the optional responses had no value or there were no responses at all, as either we were trying to retrieve those when any, or we were setting it's value to `None` if not found, instead of `{"user_id": None, "values": None, "status": None}` which is what 🤗 Datasets is producing based on the features.

So on, this PR solves that bug and now the flow mentioned above is possible and works as expected!

**Type of change**

- [X] Bug fix (non-breaking change which fixes an issue)

**How Has This Been Tested**

- [X] To also cover the scenario when there are no responses, the fixture `feedback_dataset_records` has been modified to contain a record with and another without responses

**Checklist**

- [X] I have merged the original branch into my forked branch
- [X] follows the style guidelines of this project
- [X] I did a self-review of my code
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

alvarobartt committed:
fix: append `responses` if any

alvarobartt committed:
test: `FeedbackRecord` with no responses fixture

alvarobartt committed:
docs: update `CHANGELOG.md`

alvarobartt committed:
docs: rename `issues` to `pull`

dvsrepo reviewed (commented):
cool @alvarobartt ! please ask someone to review this as I might just approve it blindly 😄

frascuchon commented on src/argilla/client/feedback/dataset.py:
```diff
@@ -759,13 +759,12 @@ def format_as(self, format: Literal["datasets"]) -> "Dataset":
-                                "value": r.values[question.name].value,
-                                "status": r.status,
+                                "user_id": r.user_id or None,
+                                "value": r.values[question.name].value or None,
```
this may fail when the value is a `0`. In those cases, the returning value will be `None, ' which is unexpected. Can you be more explicit in setting the default value for user_id, value, and status?

alvarobartt replied:
Nice catch, didn't think about the `0 or None` case, but you're right!

frascuchon commented on tests/client/conftest.py:
```diff
@@ -428,17 +428,6 @@ def feedback_dataset_records() -> List[FeedbackRecord]:
         ),
         FeedbackRecord(
             fields={"text": "This is a negative example", "label": "negative"},
-            responses=[
-                {
-                    "values": {
-                        "question-1": {"value": "This is a response to question 1"},
-                        "question-2": {"value": 1},
-                        "question-3": {"value": "a"},
-                        "question-4": {"value": ["a", "b"]},
-                    },
-                    "status": "submitted",
-                }
-            ],
```
I think we should include a specific test case to check the bugfix. Not sure this change is covering it.

alvarobartt replied:
This covers it, as well as ensuring robustness since the responses are optional, so we're mocking a `FeedbackRecord` with and without responses 👍🏻

frascuchon commented on src/argilla/client/feedback/dataset.py:
```diff
@@ -759,13 +759,12 @@ def format_as(self, format: Literal["datasets"]) -> "Dataset":
-                                "value": r.values[question.name].value,
-                                "status": r.status,
+                                "user_id": r.user_id or None,
+                                "value": r.values[question.name].value or None,
```
Since some questions can be optional, I think we should prevent this.
```suggestion
                                "value": r.values.get(question.name) or None, # TODO: we should fetch first the response and then the value
```

alvarobartt committed:
fix: `format_as("datasets")` responses

alvarobartt committed:
fix: `push_to_huggingface` parsing of UUIDs (#3231)

# Description

Due to the recent constraint for the `FeedbackRecord.user_id`s to be
`UUID`s instead of `str`s, the conversion to `dict()` was failing, as
the `UUID`s are not JSON-serializable, which was leading to some issues
when trying to add the example record to the `DatasetCard` via the
`dict()` conversion.

**Type of change**

- [X] Bug fix (non-breaking change which fixes an issue)

**How Has This Been Tested**

- [X] Re-run unit tests to pass when generating the `DatasetCard` for
`FeedbackDataset`s

**Checklist**

- [X] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [X] follows the style guidelines of this project
- [X] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

alvarobartt committed:
docs: update `CHANGELOG.md`
